plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseJdbcMetadataFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.clickhouse;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import io.trino.plugin.jdbc.DefaultJdbcMetadataFactory;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcMetadata;
+import io.trino.plugin.jdbc.JdbcQueryEventListener;
+
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+public class ClickHouseJdbcMetadataFactory
```

Comment:
We need to make it easy to disable JDBC connectors without overriding `JdbcMetadataFactory`, so one can only disable it in `JdbcClient`

Reply:
he problem is that we we first perform check in QueryPlanner.createMergePipeline() - `metadata.getRowChangeParadigm`
And if it's not implemented we break flow.

If we rely on `JdbcClient`, we will fail during `applyUpdate()` much later.

If you think it's ok I can try to refactor and adjust tests (most of them expect to fail on `getRowChangeParadigm`

Reply:
> the problem is that we we first perform check in QueryPlanner.createMergePipeline() - metadata.getRowChangeParadigm

That can also call something on `JdbcClient` to verify if it is supported.

Reply:
Overriding `JdbcMetadataFactory` is only an option for very custom connectors. Here it is common that connector may support update (or not)

Reply:
> Overriding `JdbcMetadataFactory` is only an option for very custom connectors. Here it is common that connector 
may support update (or not)

sure but cannot figure out good way `getRowChangeParadigm` is used to determine that we actually support update (not only metadata update).

What you think about, that to disable it for all connectors and enable it with overriding `JdbcMetadataFactory` for connectors we want to support update ?

Reply:
As it was done before for Kudu and others

Reply:
> If we rely on `JdbcClient`, we will fail during `applyUpdate()` much later.

sorry not on JDBCClient but on` Metadata.applyUpdate`

Reply:
Ok let me try to do disable it only by overriding JDBCClient.
It should work I think.
The only problem that we have bunch of tests, which rely on fact that we will fail during `getRowChangeParadigm`

Reply:
But maybe it would be not so dramatic